packages/world/src/World.sol

```diff
@@ -52,7 +52,7 @@ contract World is StoreData, IWorldKernel {
   /**
    * @dev Prevents the World contract from calling itself.
    */
-  modifier requireNoCallback() {
```

Comment:
not blocking but I feel like we could expand the docs here to explain _why_ this modifier is here and important to be used, esp since this relates to a vulnerability

Reply:
added issue here: https://github.com/latticexyz/mud/issues/2106